core/src/main/scala/org/apache/spark/network/BlockDataManager.scala

```diff
@@ -32,6 +32,12 @@ trait BlockDataManager {
    */
   def getBlockData(blockId: BlockId): ManagedBuffer
 
+  /**
+   * Interface to get shuffle block data. Throws an exception if the block cannot be found or
+   * cannot be read successfully.
+   */
+  def getShuffleBlockData(blockId: BlockId, shuffleGenerationId: Int): ManagedBuffer
```

Comment:
nit: `shuffleGenerationId` should be put before `blockId`.

Reply:
BTW, can we just define it as `def getShuffleBlockData(shuffleId: Int, shuffleGenerationId: Int, mapId: Int, reducerId: Int)`?

Reply:
and why doesn't the `putBlockData` method need a `shuffleGenerationId` parameter?

Reply:
```
nit: shuffleGenerationId should be put before blockId.
```
Copy that, will be done in next commit and also change the order in `ShuffleBlockResolver.getBlockData`

```
BTW, can we just define it as def getShuffleBlockData(shuffleId: Int, shuffleGenerationId: Int, mapId: Int, reducerId: Int)?
```
Yep, it's more natural when it is used in NettyBlockRpcServer, but in ShuffleBlockFetcherIterator, the blockId will be unapply and apply. WDYT?

```
why doesn't the putBlockData method need a shuffleGenerationId parameter?
```
I also spent some time on this problem during this work. [This comment](https://github.com/apache/spark/pull/24892#discussion_r296475671) contains the partial answer. In the `BlockDataManager`, as the original design, getBlockData and putBlockData are pair operation response for saving and getting local block data. The shuffle block data is controlled by ShuffleBlockResolver, ShuffleWriter, and MapOutputTracker, which didn't use putBlockData.
So as we discuss in the above comment, after deleting the old version support in a later version, we'll get the cleanest semantic for these 3 operations, get/putBlockData only response for local block and getShuffleBlockData handle shuffle block data fetching.